go-ipfs fails to build on i386

<!-- Output From `ipfs version --all` -->
#### Version information:

5c9125e8b55bc7d9e60f8a9169473da36d28b920 

<!-- Bug, Feature, Enhancement, Etc -->
#### Type: Bug

<!-- One of following:
    Critical - System crash, application panic.
    High - The main functionality of the application does not work, API breakage, repo format breakage, etc.
    Medium - A non-essential functionality does not work, performance issues, etc.
    Low - An optional functionality does not work.
    Very Low - Translation or documentation mistake. Something that really does not matter much but should be noticed for a future release. -->
#### Severity: Critical

#### Description: 

```
# gx/ipfs/QmTBxwy8cerzXbZQFUwTBCSxx55jUgVzudFcpmnAHUGuPy/badger
../../../gx/ipfs/QmTBxwy8cerzXbZQFUwTBCSxx55jUgVzudFcpmnAHUGuPy/badger/kv.go:86:6: constant 1099511627776 overflows int
cmd/ipfs/Rules.mk:21: recipe for target 'cmd/ipfs/ipfs' failed
make: *** [cmd/ipfs/ipfs] Error 2
```

--- a/package.json
+++ b/package.json
@@ -460,9 +460,9 @@
     },
     {
       "author": "magik6k",
-      "hash": "QmUamAGkvPp1w84dfc2YMy9ic6iyBvaRoaTiaat8Crtawq",
+      "hash": "QmNm2bfBCtEzmpxVzaW2FZGzWx2KWHprfd27aWoXptrAGa",
       "name": "go-ds-badger",
-      "version": "0.4.1"
+      "version": "1.0.4"
     },
     {
       "author": "whyrusleeping",

--- a/repo/fsrepo/datastores.go
+++ b/repo/fsrepo/datastores.go
@@ -16,8 +16,8 @@ import (
 	ds "gx/ipfs/QmVSase1JP7cq9QkPT46oNwdp9pT6kBkG3oqS14y3QcZjG/go-datastore"
 	mount "gx/ipfs/QmVSase1JP7cq9QkPT46oNwdp9pT6kBkG3oqS14y3QcZjG/go-datastore/syncmount"
 
+	badgerds "gx/ipfs/QmNm2bfBCtEzmpxVzaW2FZGzWx2KWHprfd27aWoXptrAGa/go-ds-badger"
 	levelds "gx/ipfs/QmPdvXuXWAR6gtxxqZw42RtSADMwz4ijVmYHGS542b6cMz/go-ds-leveldb"
-	badgerds "gx/ipfs/QmUamAGkvPp1w84dfc2YMy9ic6iyBvaRoaTiaat8Crtawq/go-ds-badger"
 	ldbopts "gx/ipfs/QmbBhyDKsY4mbY6xsKt3qu9Y7FPvMJ6qbD8AMjYYvPRw1g/goleveldb/leveldb/opt"
 )
 

